examples/technical_blogs/tartan_blog/build_and_run_networks.py

```diff
@@ -0,0 +1,156 @@
+    
+    create_directory(portBlas_build_dir_path)
+    os.chdir(portBlas_build_dir_path)
+    
+    cmake_portBlas_command = [
+        "cmake",
+        f"{portBLAS_source}",
+        "-GNinja",
+        "-DSYCL_COMPILER=dpcpp"
+    ]
+
+    run_cmake_and_ninja(cmake_portBlas_command)
+    os.chdir("..")
+
+# Build SYCL-DNN
+def build_sycl_dnn():
+    script_directory = os.getcwd()
+    sycl_dnn_build_dir = os.path.join(script_directory, "build_sycl_dnn")
+    
+    create_directory(sycl_dnn_build_dir)
+    os.chdir(sycl_dnn_build_dir)
+    
+    cmake_sycl_dnn_command = [
+        "cmake",
+        f"{SYCL_DNN_source}",
+        "-GNinja",
+        f"-DCMAKE_CXX_COMPILER={os.environ['CMAKE_CXX_COMPILER']}",
+        "-DSNN_BUILD_BENCHMARKS=OFF",
+        "-DSNN_BENCH_SYCLBLAS=OFF",
+        "-DSNN_BUILD_DOCUMENTATION=OFF",
+        f"-DSyclBLAS_DIR='{os.path.join(script_directory, 'build_portBlas')}'"
+    ]
+
+    run_cmake_and_ninja(cmake_sycl_dnn_command)
+    os.chdir("..")
+
+
+def download_and_convert_model(file_url, destination_file, h5_to_bin_script):
+    try:
+        response = requests.get(file_url)
+        response.raise_for_status()
+        with open(destination_file, "wb") as file:
+            file.write(response.content)
+        print(f"Downloaded '{destination_file}' successfully.")
+        
+        subprocess.run(["python", h5_to_bin_script, destination_file], check=True)
+        print(f"Converted '{destination_file}' to binary successfully.")
+    except requests.exceptions.RequestException as e:
+        print(f"Error downloading file: {e}")
+    except subprocess.CalledProcessError as e:
+        print(f"Error running h5toBin.py: {e}")
+
+
+def prepare_image(img_script_path, input_image):
+    try:
+        subprocess.run(["python", img_script_path, input_image], check=True)
+        print(f"Image prepared successfully.")
+    except subprocess.CalledProcessError as e:
+        print(f"Error executing the script: {e}")
+
+# Test on images
+def test_image(model_script, data_dir, output_file):
+    command_args = [
+        model_script,
+        data_dir + "/",
+        output_file
+    ]
+    try:
+        subprocess.run(command_args, check=True)
+        print(f"Image tested successfully with {os.path.basename(model_script)}.")
+    except subprocess.CalledProcessError as e:
+        print(f"Error executing the command: {e}")
+
+
+if __name__ == "__main__":
+    build_portBLAS()
+    build_sycl_dnn()
+    sycl_dnn_build_dir = os.path.join(os.getcwd(), "build_sycl_dnn")
+
+    # Set the environment variables
+    os.environ["CA_HAL_DEBUG"] = "1"
+    os.environ["CA_PROFILE_LEVEL"] = "3"
+    os.environ["OCL_ICD_FILENAMES"] = f"{OCK_libCL_path}"
+    os.environ["ONEAPI_DEVICE_SELECTOR"] = "opencl::acc"
+    os.environ["SYCL_CONFIG_FILE_NAME"] = "null.cfg"
+
+    # VGG16
+    vgg_dir = "vdata"
+    create_directory(vgg_dir)
+    os.chdir(vgg_dir)
```

Comment:
I generally don't like scripts changing directories as it's harder for users to recreate what's happening inside the script (e.g., by copying commands that are `echo`d to the terminal). Or maybe even the script will error, not change from `foo/bar` into `foo/bar/baz`, but will then change into `..` and end up in `foo`, not back in `foo/bar`. Then the script's in a weird state.

Here, can we set `destination_file` to to `vgg_dir/{destination_file}` and avoid changing directories? It feels as though everything will "just" work, but I don't know the nitty-gritty.